### PR TITLE
Fix install paths for cuquantum in CI and docs

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** Current install directions for cuquantum allow use of the RPATH to the expected directory with Python, but requires explicit targets when using a C++ build. The docs and paths searched for cuquantum right now often have the `lib` dir implicit in the path, which causes an incorrect path to be found by CMake, as it is a child class of the root. This PR fixes those such paths, allowing the C++ builds to find the correct custatevec libraries, ensuring a successful cmake configuration step.

**Description of the Change:** As above.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by mlxd** [comment]
Fixes current MPI-build block in https://github.com/PennyLaneAI/pennylane-lightning/pull/823

**Review by multiphaseCFD — APPROVED** [review]
Thanks @mlxd ! Looks good to me! Does it mean that there will be a scenario that `custatevec.so` located in /lib64/ dir?

**Comment by mlxd** [comment]
> Thanks @mlxd ! Looks good to me! Does it mean that there will be a scenario that `custatevec.so` located in /lib64/ dir?

No guarantee, but hard to know based on previous lib dirs I've seen in the wild.

File: cmake/support_pllgpu.cmake [hunk 1]
```diff
@@ -47,6 +47,7 @@ endmacro()
 
 # Macro to aid in finding cuStateVec lib
 macro(findCustatevec external_libs)
+    set(CQENV "$ENV{CUQUANTUM_SDK}")
```
Comment (maliasadi) [inline cmake/support_pllgpu.cmake:50]: To keep it consistent with other variables,
 ```suggestion
     set(CUQUANTUM_ENV "$ENV{CUQUANTUM_SDK}")
```

File: cmake/support_pllgpu.cmake [hunk 2]
```diff
@@ -58,6 +59,8 @@ macro(findCustatevec external_libs)
             lib64
             ${CUQUANTUM_SDK}/lib
             ${CUQUANTUM_SDK}/lib64
+            ${CQENV}/lib
+            ${CQENV}/lib64
```
Comment (maliasadi) [inline cmake/support_pllgpu.cmake:63]:
```suggestion
            ${CUQUANTUM_ENV}/lib
            ${CUQUANTUM_ENV}/lib64
```

File: cmake/support_pllgpu.cmake [hunk 3]
```diff
@@ -74,6 +77,7 @@ macro(findCustatevec external_libs)
             /opt/cuda
             include
             ${CUQUANTUM_SDK}/include
+            ${CQENV}/include
```
Comment (maliasadi) [inline cmake/support_pllgpu.cmake:80]:
```suggestion
            ${CUQUANTUM_ENV}/include
```

File: README.rst [hunk 4]
```diff
@@ -249,7 +249,7 @@ Then the `cuStateVec`_ library can be installed and set a ``CUQUANTUM_SDK`` envi
 .. code-block:: console
 
     python -m pip install wheel custatevec-cu12
-    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')")
+    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/')")
```
Comment (maliasadi) [inline README.rst:252]:
```suggestion
    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum')")
```

File: README.rst [hunk 5]
```diff
@@ -386,7 +386,7 @@ Then the `cutensornet`_ library can be installed and set a ``CUQUANTUM_SDK`` env
 .. code-block:: console
 
     pip install cutensornet-cu12
-    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')")
+    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/')")
```
Comment (maliasadi) [inline README.rst:389]:
```suggestion
    export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum')")
```

**Review by maliasadi — COMMENTED** [review]
I've never run into this issue before. nice catch! :rugby_football:

**Review by vincentmr — APPROVED** [review]
LGTM, feel free to merge when the CI is green. Cheers and thanks @mlxd !

**Review by maliasadi — APPROVED** [review]
:rugby_football:

**Review by multiphaseCFD — APPROVED** [review]
Looks good to me! Thanks @mlxd !

**Comment by mlxd** [comment]
FYI, as this flows into https://github.com/PennyLaneAI/pennylane-lightning/pull/823 I'm ignoring formatting checks until post-merge

**Pull request merged**